### PR TITLE
don't write logs which are for platforms which weren't requested or

### PR DESCRIPTION
**Pull request by rcerven**
were omitted

* CLOUDBLD-10661

Signed-off-by: Robert Cerven <rcerven@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [n/a] JSON/YAML configuration changes are updated in the relevant schema
- [n/a] Pull request has a link to an osbs-docs PR for user documentation updates

### CODE REVIEW

**Comment by rcerven** [comment]
related prs: https://github.com/containerbuildsystem/atomic-reactor/pull/1973  https://github.com/containerbuildsystem/osbs-client/pull/1109

**Comment by rcerven** [comment]
unit tests need osbs pr merged

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 1]
```diff
@@ -424,6 +425,19 @@ def _write_logs(self, build_id, logs_dir, platforms: list = None):
...
             if task_platform not in logfiles:
+                if task_platform != 'noarch' and not final_platforms:
+                    final_platforms = self.osbs().get_final_platforms(build_id)
+
+                    if not final_platforms:
+                        self.logger.info("Couldn't obtain final platforms from build")
+                        final_platforms = platforms
+
+                    else:
+                        final_platforms = final_platforms
```
Comment (lkolacek) [inline koji_containerbuild/plugins/builder_containerbuild.py:436]: what is the benefit of this assignment?

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py]: will remove it, I was modifying that assigned value before

File: koji_containerbuild/plugins/builder_containerbuild.py [hunk 2]
```diff
@@ -424,6 +425,16 @@ def _write_logs(self, build_id, logs_dir, platforms: list = None):
...
             if task_platform not in logfiles:
+                if task_platform != 'noarch' and not final_platforms:
+                    final_platforms = self.osbs().get_final_platforms(build_id)
+
+                    if not final_platforms:
+                        self.logger.info("Couldn't obtain final platforms from build")
+                        final_platforms = platforms
+
+                if (task_platform != 'noarch') and (task_platform not in final_platforms):
+                    continue
```
Comment (chmeliik) [inline koji_containerbuild/plugins/builder_containerbuild.py:436]: In atomic-reactor, logs for disabled platforms are written to osbs-build.log. Do we want to do the same thing here, or just skip them?

Comment (rcerven) [reply koji_containerbuild/plugins/builder_containerbuild.py:436]: you mean that we are logging execution of all build tasks in osbs-build.log?

well we don't have to do it here at all, as those shouldn't be in logs in koji

Comment (chmeliik) [reply koji_containerbuild/plugins/builder_containerbuild.py:436]: only the disabled ones, so there is something like `<platform> is not enabled for this build, exiting`

But yeah, we can probably just skip those here

**Review by lkolacek — APPROVED** [review]

**Review by mkosiarc — APPROVED** [review]

**Comment by rcerven** [comment]
rn updated

**Pull request merged**